packages/ui/src/components/dropdown-menu.tsx

```diff
@@ -1,199 +1,438 @@
+
+interface DropdownMenuIndicatorItemProps extends Omit<DropdownMenuItemProps, 'prefix'> {
+  color:
+    | 'gray'
+    | 'green'
+    | 'red'
+    | 'yellow'
+    | 'blue'
+    | 'purple'
+    | 'brown'
+    | 'cyan'
+    | 'indigo'
+    | 'lime'
+    | 'mint'
+    | 'orange'
+    | 'pink'
+    | 'violet'
+  pulse?: boolean
+}
+
+const DropdownMenuIndicatorItem = forwardRef<
+  ElementRef<typeof DropdownMenuPrimitive.Item>,
+  DropdownMenuIndicatorItemProps
+>(({ color, pulse, ...props }, ref) => (
+  <DropdownMenuItem
     ref={ref}
-    className={cn('px-2 py-1.5 text-sm font-semibold', inset && 'pl-8', className)}
     {...props}
+    prefix={
+      <div
+        className={cn(`cn-dropdown-menu-item-indicator cn-dropdown-menu-item-indicator-${color}`, {
+          'animate-pulse': pulse
+        })}
+      />
+    }
   />
 ))
-DropdownMenuLabel.displayName = DropdownMenuPrimitive.Label.displayName
+DropdownMenuIndicatorItem.displayName = displayNames.indicatorItem
 
-const DropdownMenuSeparator = React.forwardRef<
-  React.ElementRef<typeof DropdownMenuPrimitive.Separator>,
-  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Separator>
+const DropdownMenuSeparator = forwardRef<
+  ElementRef<typeof DropdownMenuPrimitive.Separator>,
+  ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Separator>
 >(({ className, ...props }, ref) => (
-  <DropdownMenuPrimitive.Separator
-    ref={ref}
-    className={cn('bg-cn-background-2 -mx-1 my-1 h-px', className)}
-    {...props}
-  />
+  <DropdownMenuPrimitive.Separator ref={ref} className={cn('cn-dropdown-menu-separator', className)} {...props} />
 ))
-DropdownMenuSeparator.displayName = DropdownMenuPrimitive.Separator.displayName
+DropdownMenuSeparator.displayName = displayNames.separator
 
-const DropdownMenuShortcut = ({ className, ...props }: React.HTMLAttributes<HTMLSpanElement>) => {
-  return <span className={cn('ml-auto text-2 tracking-widest opacity-60', className)} {...props} />
-}
-DropdownMenuShortcut.displayName = 'DropdownMenuShortcut'
+const DropdownMenuHeader = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) => (
+  <div className={cn('cn-dropdown-menu-header', className)} {...props} />
+)
+DropdownMenuHeader.displayName = displayNames.header
+
+const DropdownMenuFooter = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) => (
+  <div className={cn('cn-dropdown-menu-footer', className)} {...props} />
+)
+DropdownMenuFooter.displayName = displayNames.footer
+
+const DropdownMenuSpinner = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) => (
+  <div className={cn('cn-dropdown-menu-spinner', className)} {...props}>
+    <Icon className="animate-spin" name="spinner" />
+  </div>
+)
+DropdownMenuSpinner.displayName = displayNames.spinner
+
+const DropdownMenuNoOptions = ({ className, children, ...props }: HTMLAttributes<HTMLDivElement>) => (
+  <div className={cn('cn-dropdown-menu-no-options', className)} {...props}>
```

Comment:
do we need this div? can we not add this class on the `Text` directy?

Reply:
Good catch👍
We can use `Text` directly